scipy/stats/tests/test_continuous.py

```diff
@@ -550,50 +562,65 @@ def check_nans_and_edges(dist, fname, arg, res):
         assert_equal(res[outside_arg_plus], -np.inf)
-        assert_equal(res[endpoint_arg_minus & ~valid_arg], -np.inf)
-        assert_equal(res[endpoint_arg_plus & ~valid_arg], -np.inf)
+        ref = -np.inf if not is_discrete else np.inf
+        assert_equal(res[endpoint_arg_minus & ~valid_arg], ref)
+        assert_equal(res[endpoint_arg_plus & ~valid_arg], ref)
     elif fname in {'pdf'}:
         assert_equal(res[outside_arg_minus], 0)
         assert_equal(res[outside_arg_plus], 0)
-        assert_equal(res[endpoint_arg_minus & ~valid_arg], 0)
-        assert_equal(res[endpoint_arg_plus & ~valid_arg], 0)
-    elif fname in {'logcdf'}:
+        ref = 0 if not is_discrete else np.inf
+        assert_equal(res[endpoint_arg_minus & ~valid_arg], ref)
+        assert_equal(res[endpoint_arg_plus & ~valid_arg], ref)
+    elif fname in {'logcdf'} and not is_discrete:
         assert_equal(res[outside_arg_minus], -inf)
         assert_equal(res[outside_arg_plus], 0)
         assert_equal(res[endpoint_arg_minus], -inf)
         assert_equal(res[endpoint_arg_plus], 0)
-    elif fname in {'cdf'}:
+    elif fname in {'cdf'} and not is_discrete:
         assert_equal(res[outside_arg_minus], 0)
         assert_equal(res[outside_arg_plus], 1)
         assert_equal(res[endpoint_arg_minus], 0)
         assert_equal(res[endpoint_arg_plus], 1)
-    elif fname in {'logccdf'}:
+    elif fname in {'logccdf'} and not is_discrete:
         assert_equal(res[outside_arg_minus], 0)
         assert_equal(res[outside_arg_plus], -inf)
         assert_equal(res[endpoint_arg_minus], 0)
         assert_equal(res[endpoint_arg_plus], -inf)
-    elif fname in {'ccdf'}:
+    elif fname in {'ccdf'} and not is_discrete:
         assert_equal(res[outside_arg_minus], 1)
         assert_equal(res[outside_arg_plus], 0)
         assert_equal(res[endpoint_arg_minus], 1)
         assert_equal(res[endpoint_arg_plus], 0)
-    elif fname in {'ilogcdf', 'icdf'}:
+    elif fname in {'ilogcdf', 'icdf'} and not is_discrete:
         assert_equal(res[outside_arg == -1], np.nan)
         assert_equal(res[outside_arg == 1], np.nan)
         assert_equal(res[endpoint_arg == -1], a[endpoint_arg == -1])
         assert_equal(res[endpoint_arg == 1], b[endpoint_arg == 1])
-    elif fname in {'ilogccdf', 'iccdf'}:
+    elif fname in {'ilogccdf', 'iccdf'} and not is_discrete:
         assert_equal(res[outside_arg == -1], np.nan)
         assert_equal(res[outside_arg == 1], np.nan)
         assert_equal(res[endpoint_arg == -1], b[endpoint_arg == -1])
         assert_equal(res[endpoint_arg == 1], a[endpoint_arg == 1])
 
-    if fname not in {'logmean', 'mean', 'logskewness', 'skewness', 'support'}:
+    exclude = {'logmean', 'mean', 'logskewness', 'skewness', 'support'}
+    if isinstance(dist, DiscreteDistribution):
+        exclude.update({'pdf', 'logpdf'})
+
+    if (
+            fname not in exclude
+            and not (isinstance(dist, Binomial)
```

Comment:
I think something should be done about degenerate distributions to keep them from breaking tests and help reduce test complexity. For now, since it's just this one case, we skip.

Reply:
By restricting the "typical" parameters, we don't have to test degenerate distributions here, right? Then, we can add a separate test for one degenerate distribution to make sure they are ok.

Reply:
Didn't end up implementing this?